astropy/cosmology/units.py

```diff
@@ -10,10 +10,12 @@
 from astropy.units.utils import generate_unit_summary as _generate_unit_summary
 
 if TYPE_CHECKING:
-    from collections.abc import Mapping
-    from typing import Any, Literal
+    from typing import Literal
+
+    from typing_extensions import Unpack
```

Comment:
`Unpack` is used in a few places to unpack `**atzkw`. That functionality was added to `Unpack` in Python 3.12 (see [PEP 692 – Using TypedDict for more precise **kwargs typing](https://peps.python.org/pep-0692/)). I'm willing to guess that the reason for importing `Unpack` from `typing_extensions` instead of `typing` is to make that functionality available in Python 3.11. But apparently [`typing_extensions.Unpack`](https://typing-extensions.readthedocs.io/en/latest/index.html#typing_extensions.Unpack) only backports the changes to `Unpack.__repr__()`, not the ability to unpack `**kwargs`, which means that the new type annotations are valid for Python 3.12 and later, but not for Python 3.11.

Reply:
`Unpack` has been in typing since 3.11.

Reply:
The fixes to repr are 3.12 and typing_extensions back ported.

Reply:
In Python 3.11 [`typing.Unpack`](https://docs.python.org/3.11/library/typing.html#typing.Unpack) can only be used to unpack tuples.

Reply:
Then does it come through as an Any? My mypy isn't flagging any issues.

Reply:
I haven't tested this with Python 3.11.

Reply:
It is possible that I am misunderstanding the `typing_extensions` documentation and this is actually not a problem. Someone should type check this with Python 3.11 to find out for sure.

Reply:
I looked into it more. It doesn't do it. Sigh. I kind of don't want to change it, but should. Just defeats the point of adding `_ZAtValueKWArgs`

Reply:
Working around it the gross way.

Reply:
That's not the prettiest code, but it has no effect at runtime, we only need it until we drop support for Python 3.11 and Ruff will not let us forget about it when we do.